tsuru/client/deploy.go

```diff
@@ -290,13 +366,36 @@ func targz(ctx *cmd.Context, destination io.Writer, filepaths ...string) error {
 		if err != nil {
 			return err
 		}
+		wd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		fiName := filepath.Join(wd, fi.Name())
 		if fi.IsDir() {
```

Comment:
I think lines 375-383 can be moved to before the `if` statement (so you can remove the ones in 389-397).